rotkehlchen/tests/unit/test_assets.py

```diff
@@ -935,3 +939,118 @@ def test_all_assets_pagination(globaldb: 'GlobalDBHandler', database: 'DBHandler
+    for asset_symbol, usd_value in assets.items():
+        balances.append(DBAssetBalance(
+            category=BalanceType.ASSET,
+            time=timestamp,
+            asset=Asset(asset_symbol),
+            amount=FVal('1'),
+            usd_value=usd_value,
+        ))
+        GlobalDBHandler.add_user_owned_assets([Asset(asset_symbol)])
+
+    with database.conn.write_ctx() as write_cursor:
+        database.add_multiple_balances(write_cursor, balances)
+
+    # Test case 1: Threshold $5 (basic threshold test)
+    filter_query = AssetsFilterQuery.make(
+        hide_assets_threshold=FVal('5'),
+        show_user_owned_assets_only=True,
+    )
+    results, count = GlobalDBHandler.retrieve_assets(database, filter_query)
+    assert count == len(results)
+    assert count == 3
+    assert {r['identifier'] for r in results} == {'LTC', 'XDAI', 'ETH'}
+
+    # Test case 2: High threshold ($1000) should only show high-value assets
+    filter_query = AssetsFilterQuery.make(
+        hide_assets_threshold=FVal('1000'),
+        show_user_owned_assets_only=True,
+    )
+    results, count = GlobalDBHandler.retrieve_assets(database, filter_query)
+    assert count == len(results)
+    assert count == 2
+    assert {r['identifier'] for r in results} == {'LTC', 'ETH'}
+
+    # Test case 3: Very low threshold ($0.1) should show all assets
+    filter_query = AssetsFilterQuery.make(
+        hide_assets_threshold=FVal('0.1'),
+        show_user_owned_assets_only=True,
+    )
+    results, count = GlobalDBHandler.retrieve_assets(database, filter_query)
+    assert count == len(results)
+    assert count == 5
+    assert {r['identifier'] for r in results} == {'LTC', 'XRP', 'XDAI', 'ETH', 'USD'}
+
+    # Test case 4: No threshold (should show all assets)
+    filter_query = AssetsFilterQuery.make(
+        hide_assets_threshold=None,
+        show_user_owned_assets_only=True,
+    )
+    results, count = GlobalDBHandler.retrieve_assets(database, filter_query)
+    assert count == len(results)
+    assert count == 5
+    assert {r['identifier'] for r in results} == {'LTC', 'XRP', 'XDAI', 'ETH', 'USD'}
+
+
+def test_assets_threshold_usd_value_filtering_with_ignored_assets(database):
+    """Test that threshold filtering works correctly with ignored assets"""
+
+    timestamp = ts_now()
+
+    # Add some test assets
+    assets = {
+        'LTC': FVal('2000'),
+        'XRP': FVal('5'),
+        'XDAI': FVal('10'),
+    }
+
+    # Clear existing balances if any and adding ignored asset
+    with database.conn.write_ctx() as write_cursor:
+        write_cursor.execute('DELETE FROM timed_balances')
+        database.add_to_ignored_assets(write_cursor, Asset('XDAI'))
+
+    timestamp = ts_now()
+    balances = []
+    for asset_symbol, usd_value in assets.items():
+        balances.append(DBAssetBalance(
+            category=BalanceType.ASSET,
+            time=timestamp,
+            asset=Asset(asset_symbol),
+            amount=FVal('1'),
```

Comment:
just as a note instead of using `FVal('1')` or `FVal(0)` we use `ONE` and `ZERO` that are predefined constants